modules/pam_env/pam_env.c

```diff
@@ -311,7 +311,7 @@ static int _assemble_line(FILE *f, char *buffer, int buf_len)
 	    D(("_assemble_line: corrupted or binary file"));
 	    return -1;
 	}
-	if (p[strlen(p)-1] != '\n') {
+	if (p[strlen(p)-1] != '\n' && !feof(f)) {
```

Comment:
I don't think this would work, feof wouldn't return nonzero until fgets returned NULL.
I suppose you need a second fgets call.

Reply:
No, at least according to my testing if fgets() returns because there is no EOL at the EOF, the feof() will return 1 immediately. There is no need for another fgets().

Reply:
OK then.